src/settings_gui.cpp

```diff
@@ -890,14 +997,27 @@ struct GameOptionsWindow : Window {
 
 	void OnMouseLoop() override
 	{
-		if (_left_button_down || this->gui_scale == _gui_scale) return;
+		if (_left_button_down) {
+			return;
+		}
 
-		_gui_scale_cfg = this->gui_scale;
+		if (this->gui_scale != _gui_scale)
+		{
+			_gui_scale_cfg = this->gui_scale;
 
-		if (AdjustGUIZoom(false)) {
-			ReInitAllWindows(true);
-			this->SetWidgetLoweredState(WID_GO_GUI_SCALE_AUTO, false);
-			this->SetDirty();
+			if (AdjustGUIZoom(false)) {
+				ReInitAllWindows(true);
+				this->SetWidgetLoweredState(WID_GO_GUI_SCALE_AUTO, false);
+				this->SetDirty();
+			}
+		} else if (static_cast<uint>(this->small_font_size) != _fcsettings.small.size) {
+			ResizeFont(FS_SMALL, static_cast<uint>(this->small_font_size));
+		} else if (static_cast<uint>(this->medium_font_size) != _fcsettings.medium.size) {
+			ResizeFont(FS_NORMAL, static_cast<uint>(this->medium_font_size));
+		} else if (static_cast<uint>(this->large_font_size) != _fcsettings.large.size) {
+			ResizeFont(FS_LARGE, static_cast<uint>(this->large_font_size));
+		} else if (static_cast<uint>(this->monospaced_font_size) != _fcsettings.mono.size) {
+			ResizeFont(FS_MONO, static_cast<uint>(this->monospaced_font_size));
 		}
```

Comment:
I suspect it's not actually possible to trigger more than one of these conditions at once, but maybe it would work better conceptually if it wasn't just a massive if/else if chain?

Reply:
It shouldn't be possible.